app_feup/lib/view/Pages/login_page_view.dart

```diff
@@ -153,6 +155,33 @@ class _LoginPageViewState extends State<LoginPageView> {
     );
   }
 
+  /// Creates the widget for the user to choose their faculty
+  Widget createFacultyInput() {
+    final faculties = ['faup', 'fbaup', 'fcup', 'fcnaup', 'fadeup', 'fdup',
+      'fep', 'feup', 'ffup', 'flup', 'fmup', 'fmdup', 'fpceup', 'icbas'];
+
+    return DropdownButton<String> (
+      value: faculty,
+      items: faculties.map<DropdownMenuItem<String>>((String value) {
+        return DropdownMenuItem<String> (
+          value: value,
+          child: Text(value),
+        );
+      }).toList(),
+      onChanged: (newDropdownValue) {
+        setState(() {
+          faculty = newDropdownValue;
+        });
+      },
+      isExpanded: true,
+      dropdownColor: Color.fromARGB(255, 0x75, 0x17, 0x1e),
+      iconDisabledColor: Colors.white,
+      iconEnabledColor: Colors.white,
+      style: TextStyle(color: Colors.white, fontSize: 20),
+      underline: Container(width: 200, height: 0.2, color: Colors.black87),
+    );
+  }
```

Comment:
Types don't need to be indicated when they can be inferred. We try to make use of that to reduce unnecessary code to understand its fundamentals:
```suggestion
    return DropdownButton (
      value: faculty,
      items: faculties.map((String value) {
        return DropdownMenuItem (
          value: value,
          child: Text(value),
        );
      }).toList(),
      onChanged: (newDropdownValue) {
        setState(() {
          faculty = newDropdownValue;
        });
      },
      isExpanded: true,
      dropdownColor: Color.fromARGB(255, 0x75, 0x17, 0x1e),
      iconDisabledColor: Colors.white,
      iconEnabledColor: Colors.white,
      style: TextStyle(color: Colors.white, fontSize: 20),
      underline: Container(width: 200, height: 0.2, color: Colors.black87),
    );
  }
```

Reply:
> I'm not familiarized with the rest of the code yet, is it ok to make the change? If not, is there a way we can have the items in uppercase but still store the value in faculty as a lowecase string?

I think the easiest way would be to change the values that are being displayed, without actually changing the strings themselves, such as this: 
```suggestion
return DropdownButton (
      value: faculty.toUpperCase(),
      items: faculties.map((value) =>
        DropdownMenuItem (
          value: value,
          child: Text(value.toUpperCase()),
        )
      ).toList(),
      onChanged: (newDropdownValue) {
        setState(() {
          faculty = newDropdownValue;
        });
      },
      isExpanded: true,
      dropdownColor: Theme.of(context).accentColor,
      iconDisabledColor: Colors.white,
      iconEnabledColor: Colors.white,
      style: TextStyle(color: Colors.white, fontSize: 20),
      underline: Container(width: 200, height: 0.2, color: Colors.black87),
    );
```

Reply:
Thanks! That makes much more sense!